packages/functions/lib/index.d.ts

```diff
@@ -144,6 +144,30 @@ export namespace FirebaseFunctionsTypes {
     (data?: any): Promise<HttpsCallableResult>;
   }
 
+  /**
+   * An HttpsCallableOptions object that can be passed as the second argument to `firebase.functions().httpsCallable(name, HttpsCallableOptions)`.
+   **/
+  export interface HttpsCallableOptions {
+    /**
+     * The timeout property allows you to control how long the application will wait for the cloud function to respond in milliseconds.
+     *
+     * #### Example
+     *
+     *```js
+     * // The below will wait 7 seconds for a response from the cloud function before an error is thrown
+     * try {
+     *  const instance = firebase.functions().httpsCallable('order', { timeout: 7000 });
+     *  const response = await instance({
+     *    id: '12345',
+     *  });
+     * } catch (e) {
+     *  console.log(e);
+     * }
+     * ```
+     */
+    timeout?: number;
```

Comment:
The JS SDK is in millis https://github.com/firebase/firebase-js-sdk/blob/74870c7a3832ec0c1e6f93e4d45e9d0a3180dfb3/packages/functions-types/index.d.ts#L37

The iOS SDK PR that implemented it says it will be 60 seconds in the code but is actually 70 after a commit in the PR changed it from 60 to 70 without comment https://github.com/firebase/firebase-ios-sdk/pull/2329/files and is in seconds not millis (double 🤦 )

Android also does 70 seconds and in units of seconds https://github.com/firebase/firebase-android-sdk/blob/master/firebase-functions/src/main/java/com/google/firebase/functions/HttpsCallOptions.java

Can't believe they did that (seconds vs millis). Thank you in my case at least for using milliseconds so that it maintains JS consistency and works for me on web and mobile

Reply:
JS SDK consistency is our motto